### DISCUSSION TITLE
Window focus via mod+arrow broken in 2x2 grid

### DISCUSSION THREAD

**Discussion by MartyMcFlyInTheSky** [i3/i3#5849] [Q&A] [answered]
When I'm shifting focus via up down left right arrow keys + mod in a 2x2 grid (say all terminals), it doesn't work as expected for me for the lower two windows. Usually when Im focusing right while focus is on lower left window, the upper right window is selected and vice versa. Sometimes it works but it depends on which window was selected first... Does this have to do with the internal tree representation of i3 windows?

In any case, can i fix it so that i get a more visually intuitive experience for focus shift?

**Comment by okraits** [comment]
This is intentional. If you have several stacks of windows next to each other, i3 remembers the last focused window in each stack and focuses that if you move the focus across the stacks.

**Reply by MartyMcFlyInTheSky** [reply]
Uh. I almost never use stacks. Can I change this behaviour then?

**Reply by orestisfl** [reply, marked as answer]
The behavior is not unique to stacks, it applies to all layouts.

> Does this have to do with the internal tree representation of i3 windows?

Yes. i3 doesn't care about the geometric positioning of the windows, it uses the tree to traverse and decide which window is focused next. This behavior is not configurable.

**Comment by okraits** [comment]
> Uh. I almost never use stacks. Can I change this behaviour then?

I wasn't talking about the stacked layout. With stacks I meant several columns of one or more windows above each other. No, this can't be changed as it is intentional.
